src/Pux/Router.purs

```diff
@@ -154,13 +158,19 @@ instance matchApplicative :: Applicative Match where
   pure a = Match \r -> pure $ Tuple r a
 
 routeFromUrl :: String -> Route
-routeFromUrl url | url == "/" = Nil
-                 | otherwise = map parsePart $ drop 1 $ fromFoldable (S.split "/" url)
```

Comment:
bug - the query is never separated (if the url does not end in /), which is the case if it ends in index.html for example.